networkit/test/test_graph.py

```diff
@@ -119,7 +119,115 @@ def nodeInIter(node):
+		up2 = nk.dynamics.GraphEvent(nk.dynamics.GraphEventType.NODE_ADDITION, 1, 1, 1)
+		self.assertNotEqual(up1,up2)
+		self.assertLess(up1,up2)
+		self.assertLessEqual(up1,up2)
+		self.assertGreater(up2,up1)
+		self.assertGreaterEqual(up2,up1)	
+
+	def testGraphFromGraphEventStream(self):
+		up1 = nk.dynamics.GraphEvent(nk.dynamics.GraphEventType.NODE_ADDITION, 0, 0, 0)	
+		up2 = nk.dynamics.GraphEvent(nk.dynamics.GraphEventType.NODE_ADDITION, 1, 1, 1)	
+		up3 = nk.dynamics.GraphEvent(nk.dynamics.GraphEventType.NODE_ADDITION, 2, 2, 2)		
+		up4 = nk.dynamics.GraphEvent(nk.dynamics.GraphEventType.NODE_ADDITION, 3, 3, 3)	
+		upE1 = nk.dynamics.GraphEvent(nk.dynamics.GraphEventType.EDGE_ADDITION, 1, 2, 2.0)	
+		upE2 = nk.dynamics.GraphEvent(nk.dynamics.GraphEventType.EDGE_ADDITION, 2, 3, 3.0)	
+		upE3 = nk.dynamics.GraphEvent(nk.dynamics.GraphEventType.EDGE_ADDITION, 1, 3, 1.0)	
+		allUpdates=[up1,up2,up3,up4,upE1,upE2,upE3]
+		G = nk.dynamics.graphFromStream(allUpdates, weighted=True,directed=True)
+		self.assertEqual(G.numberOfNodes(), 4)
+		self.assertEqual(G.numberOfEdges(), 3)		
+	
+	def testGraphDifference(self):
+		G1 = self.getSmallGraph(weighted=True,directed=True)
+		G2 = self.getSmallGraph(weighted=True,directed=True)
+		update1 = nk.dynamics.GraphEvent(nk.dynamics.GraphEventType.EDGE_ADDITION, 0, 3, 2.0)	
+		update2 = nk.dynamics.GraphEvent(nk.dynamics.GraphEventType.EDGE_ADDITION, 2, 3, 1.0)	
+		updateG2 = nk.dynamics.GraphUpdater(G2)
+		updateG2.update([update1,update2])
+		GraphDiff = nk.dynamics.GraphDifference(G1,G2)
+		GraphDiff.run()
+		#testing graphDifference functions
+		self.assertEqual(len(GraphDiff.getEdits()), 2)
+		self.assertEqual(GraphDiff.getNumberOfEdits(), 2)
+		self.assertEqual(GraphDiff.getNumberOfNodeAdditions(), 0)
+		self.assertEqual(GraphDiff.getNumberOfNodeRestorations(), 0)
+		self.assertEqual(GraphDiff.getNumberOfNodeRemovals(), 0)
+		self.assertEqual(GraphDiff.getNumberOfEdgeAdditions(), 2)					
+		self.assertEqual(GraphDiff.getNumberOfEdgeRemovals(), 0)
+		self.assertEqual(GraphDiff.getNumberOfEdgeWeightUpdates(), 0)
+	
+	def testGraphPickling(self):
+		G = nk.Graph(2)
+		G.addEdge(0,1)
+		G.indexEdges()
+		pickledGraph = pickle.dumps(G)
+		G2 = pickle.loads(pickledGraph)
+		self.assertEqual(G.numberOfNodes(), G2.numberOfNodes())
+		self.assertEqual(G.numberOfEdges(), G2.numberOfEdges())
+		self.assertEqual(G.edgeId(0,1), G2.edgeId(0,1))			
+
+	def testNodeIterator(self):
+		nk.setSeed(42, False)
+
+		g = self.getSmallGraph()
+
+		def doTest(g):
+			nodes = []
+			g.forNodes(lambda u: nodes.append(u))
+
+			i = 0
+			for u in g.iterNodes():
+				self.assertEqual(u, nodes[i])
+				i += 1
+
+		doTest(g)
+		g.removeNode(nk.graphtools.randomNode(g))
+		g.removeNode(nk.graphtools.randomNode(g))
+		doTest(g)	
+	
+	def testNodesInRandomOrder(self):
+		G = nk.Graph(3, directed=True)
+		G.addEdge(0,1)
+		G.addEdge(0,2)
+		G.addEdge(1,2)
+		G.forNodesInRandomOrder(lambda u: self.assertFalse(G.hasEdge(u, 0)))
```

Comment:
I don't follow the logic of this test. Shouldn't we check that the node `u` exists and that the iterator goes over all nodes?